### DISCUSSION TITLE
How do I lookup an arbitrary session?

### DISCUSSION THREAD

**Discussion by paambaati** [remix-run/remix#7570] [Q&A] [answered]
For debugging puposes, as an admin, I'd like to lookup some other user's session – provided they give me their session ID. Because session implementations are deeply linked with cookies, I'm struggling to both construct the cookie for a given session ID (after which I can `getSession` on that cookie) or directly lookup a session with `getSession` with an `id` instead of a cookie.

How does one go about doing this?

**Comment by sergiodxa** [comment]
What session storage are you using? If you used createCookieSessionStorage then the cookie is the session, you will need to duplicate the session content in that case.

If you use another session storage mechanism the cookie will only contain a session ID that you can use to find the session data on another storage, this other storage can be anything from the file system to your DB.

**Reply by paambaati** [reply, marked as answer]
I use disk and redis storage, and I ended up doing this –

```typescript
//disk.storage.server.ts
import { getFile } from "@remix-run/node/dist/sessions/fileStorage";

export const lookupDiskSessionByID = async (
  id: string,
): Promise<string | null> => {
  const filename = getFile(SESSIONS_DIRECTORY, id);  // <- main thing.
  try {
    const fileContent = await readFileAsync(filename);
    return fileContent;
  } catch {
    return null;
  }
};
```

```typescript
//redis.storage.server.ts
export const lookupRedisSessionByID = async (
  id: string,
): Promise<string | null> => {
  const redisUrl = environment().REDIS_URL as string;
  const redis = getRedisInstance(redisUrl);
  return await redis.get(id); // <- main thing.
};

```
